app/uk/gov/hmrc/gform/eval/EvaluationResults.scala

```diff
@@ -508,6 +547,23 @@ case class EvaluationResults(
           case _              => unsupportedOperation("String")(expr)
         }
       case RemoveSpaces(fcId) => removeSpaces(fcId, recData)
+      case NumberedList(fcId)
+          if evaluationContext.addressLookup(fcId.baseComponentId) || evaluationContext
+            .overseasAddressLookup(fcId.baseComponentId) =>
+        val list = atlAddresses(fcId)
+        StringResult(govukList(list, false))
+      case NumberedList(fcId) =>
+        val list = atlValues(fcId)
+        StringResult(govukList(list, false))
+      case BulletedList(fcId)
+          if evaluationContext.addressLookup(fcId.baseComponentId) || evaluationContext
+            .overseasAddressLookup(fcId.baseComponentId) =>
+        val list = atlAddresses(fcId)
```

Comment:
This special handling of addresses should be eliminated and replaced by `loop(FormCtx(fcId))` which will return `ListResult`.
Unfortunately printing addresses from ATL into smartstrings is not currently implemented, but feel free to fix it.

Reply:
yes, that's why I implemented it separately. I will look into fixing it :+1: